### DISCUSSION TITLE
How to develop controls for Avalonia?

### DISCUSSION THREAD

**Discussion by arivoir** [AvaloniaUI/Avalonia#15599] [Q&A] [answered]
Hi,
I'm a WPF(and other frameworks) control developer, and I'm wondering how well the controls could work on other platforms but windows.
I just created a new empty Avalonia app using the template, visualized it in vscode, and run it on mac successfully.
Now I tried to add a control library to it but having build issues.

```
  <ItemGroup>
    <PackageReference Include="C1.WPF.Grid" Version="8.0.20233.668" />
  </ItemGroup>
```

_error NETSDK1073: The FrameworkReference 'Microsoft.WindowsDesktop.App.WPF' was not recognized_

Will Avalonia work with a reference to a package built for _net8.0-windows_? Does the controls need to be rebuilt for Avalonia? 

I appreciate you guidance.

**Comment by workgroupengineering** [comment, marked as answer]
You have to use [XPF](https://avaloniaui.net/XPF).

**Comment by arivoir** [comment]
Is it possible to create controls for Avalonia UI instead of XPF?

**Reply by timunie** [reply]
Sure, just check out Avalonia.Samples and Awsome-Avalonia. Both can be found on GitHub. Moreover as we are open source, you can explore our controls as well. 

Welcome 🤗

**Comment by thevortexcloud** [comment]
To elaborate: Avalonia is not WPF. It's very similar though. If you actually want to use WPF you need to either embed a WPF app inside Avalonia or use XPF. Otherwise you need to make a native Avalonia control. The process for doing that is pretty much exactly the same as it is in WPF, although you will run into some differences.
